From 08aa9a26779d4ede3de7857fafd03bd6331ba948 Mon Sep 17 00:00:00 2001
From: Philippe Proulx <eeppeliteloop@gmail.com>
Date: Mon, 6 Nov 2017 15:46:03 -0500
Subject: [PATCH] configure.ac: add --disable-examples option to not
 build/install examples

Some environments and distributions do not need the LTTng-UST examples
to be built because they remove them anyway. Continue to build them by
default, but add --disable-examples to explicitly disable them.

Signed-off-by: Philippe Proulx <eeppeliteloop@gmail.com>
[Philippe: backport from upstream commit 08aa9a26]
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 configure.ac    | 16 ++++++++++++++++
 doc/Makefile.am |  6 +++++-
 2 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 10087f3f..68f41315 100644
--- a/configure.ac
+++ b/configure.ac
@@ -423,6 +423,19 @@ AC_DEFINE_UNQUOTED([LTTNG_SYSTEM_RUNDIR], ["$lttng_system_rundir"],
 AC_CHECK_PROG([BUILD_GEN_TP_EXAMPLES], [python], ["yes"])
 AM_CONDITIONAL([BUILD_GEN_TP_EXAMPLES], [test "x$BUILD_GEN_TP_EXAMPLES" = "xyes"])

+# Enable building examples
+AC_ARG_ENABLE(
+	examples,
+	AS_HELP_STRING(
+		[--disable-examples],
+		[Do not build and install examples]
+	),
+	[enable_examples=$enableval],
+	[enable_examples=yes]
+)
+
+AM_CONDITIONAL([ENABLE_EXAMPLES], [test "x$enable_examples" != "xno"])
+
 # Set $IN_GIT_REPO if we're in the Git repository; the `bootstrap` file
 # is not distributed in tarballs.
 AS_IF([test -f "$srcdir/bootstrap"], [in_git_repo=yes], [in_git_repo=no])
@@ -596,6 +609,9 @@ PPRINT_PROP_BOOL([NUMA], $value)
 AS_ECHO
 PPRINT_SET_INDENT(0)

+test "x$enable_examples" = xyes && value=1 || value=0
+PPRINT_PROP_BOOL([Build and install examples], $value, $PPRINT_COLOR_SUBTITLE)
+
 # man pages build enabled/disabled
 m4_pushdef([build_man_pages_msg], [Build and install man pages])

diff --git a/doc/Makefile.am b/doc/Makefile.am
index e110ed85..a67c8c22 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,3 +1,7 @@
-SUBDIRS = . man examples
+SUBDIRS = . man
+
+if ENABLE_EXAMPLES
+SUBDIRS += examples
+endif

 dist_doc_DATA = java-agent.txt
--
2.15.0

